### PR TITLE
Move text removed by #421 to the arch document

### PR DESCRIPTION
**Pull request by philsbln**

### CODE REVIEW

**Comment by gorryfair** [comment]
NOTES 1: If this moves, please remember to rephrase the sentences to match the Arch style, content seems OK.
NOTE 2: This probably should deal with my previous issue about defining "The Socket API (Sockets)" in such as way as to define "sockets".

**Comment by philsbln** [comment]
I see value in it as it gives historic context and therefore support @theri's suggestion in #421. 
It is nothing I will really fight for, but something I wanted to try whether it is possible with little effort.

File: draft-ietf-taps-arch.md [hunk 1]
```diff
@@ -97,7 +105,13 @@ This document was developed in parallel with the specification of the Transport
...
+TCP has scaled remarkably well over the past three and a half decades, but its total ubiquity has hidden an uncomfortable fact: the network is not really a	file, and stream abstractions are too simplistic for many modern application	
+programming models.	
+
+In the meantime, the nature of Internet access, and the variety of Internet	transport protocols, is evolving. The challenges that new protocols and access paradigms present to the sockets API and to programming models based on them	
+inspire the design principles of a new approach.
+Threfore, the new Transport Services architecture is based on the survey of services provided by IETF transport protocols and congestion control mechanisms {{?RFC8095}}, and the distilled minimal set of the features offered by transport protocols {{?I-D.ietf-taps-minset}}. These documents identified common features and patterns across all transport protocols developed thus far in the IETF.
```
Comment (theri) [inline draft-ietf-taps-arch.md]: s/Threfore/Therefore/

File: draft-ietf-taps-arch.md [hunk 2]
```diff
@@ -97,7 +105,13 @@ This document was developed in parallel with the specification of the Transport
 
 ## Background
 
-The Transport Services architecture is based on the survey of services provided by IETF transport protocols and congestion control mechanisms {{?RFC8095}}, and the distilled minimal set of the features offered by transport protocols {{?I-D.ietf-taps-minset}}. These documents identified common features and patterns across all transport protocols developed thus far in the IETF.
+The BSD Unix Sockets API's (Sockets) and its SOCK_STREAM abstraction was a revolution in simplicity: by bringing network sockets into the UNIX programming model, it enabled anyone who knew how to write programs that dealt with sequential-access files to also write network applications. The simplicity of this API is a key reason the	Internet won the protocol wars {{PROTOCOL-WARS}} of the 1980s. SOCK_STREAM is	tied to the Transmission Control Protocol (TCP), specified in 1981 {{?RFC0793}}.
```
Comment (theri) [inline draft-ietf-taps-arch.md]: Why the brackets around "(Sockets)"? Without the word, the sentence wouldn't work anymore.
Also, I think we call it "Socket API", not "Sockets API" in the rest of the document.

File: draft-ietf-taps-arch.md [hunk 3]
```diff
@@ -97,7 +105,13 @@ This document was developed in parallel with the specification of the Transport
...
+The BSD Unix Sockets API's (Sockets) and its SOCK_STREAM abstraction was a revolution in simplicity: by bringing network sockets into the UNIX programming model, it enabled anyone who knew how to write programs that dealt with sequential-access files to also write network applications. The simplicity of this API is a key reason the	Internet won the protocol wars {{PROTOCOL-WARS}} of the 1980s. SOCK_STREAM is	tied to the Transmission Control Protocol (TCP), specified in 1981 {{?RFC0793}}.	
+TCP has scaled remarkably well over the past three and a half decades, but its total ubiquity has hidden an uncomfortable fact: the network is not really a	file, and stream abstractions are too simplistic for many modern application	
+programming models.	
+
+In the meantime, the nature of Internet access, and the variety of Internet	transport protocols, is evolving. The challenges that new protocols and access paradigms present to the sockets API and to programming models based on them
```
Comment (theri) [inline draft-ietf-taps-arch.md]: s/sockets API/Socket API/

File: draft-ietf-taps-arch.md (hunk 2 above)
Comment (tfpauly) [inline draft-ietf-taps-arch.md]: It also shouldn't have an apostrophe "API`s" -> "APIs"

Comment (tfpauly) [inline draft-ietf-taps-arch.md]: We also do cover some of the relationship with BSD sockets in the intro.

How much do we think we really need this history here?

**Comment by philsbln** [comment]
Hi, I guess I addressed all knits raised so far – so this PR is left with @tfpauly's question:
> How much do we think we really need this history here?

**Comment by abrunstrom** [comment]
Reading this again, I actually think we may not need the history and keeping it short may be better. We otherwise have to deal with the overlap.

File: draft-ietf-taps-arch.md [hunk 4]
```diff
@@ -89,15 +97,21 @@ This document describes an architecture for exposing transport protocol features
...
+Many application programming interfaces (APIs) to perform transport networking have been deployed, perhaps the most widely known and imitated being the BSD Socket {{POSIX}} interface (Socket API). The naming of objects and functions across these APIs is not consistent, and varies depending on the protocol being used. For example, sending and receiving streams of data is conceptually the same for both an unencrypted Transmission Control Protocol (TCP) stream and operating on an encrypted Transport Layer Security (TLS) {{?RFC8446}} stream over TCP, but applications cannot use the same socket ```send()``` and ```recv()``` calls on top of both kinds of connections. Similarly, terminology for the implementation of transport protocols varies based on the context of the protocols themselves: terms such as "flow", "stream", "message", and "connection" can take on many different meanings. This variety can lead to confusion when trying to understand the similarities and differences between protocols, and how applications can use them effectively.
```
Comment (abrunstrom) [inline draft-ietf-taps-arch.md]: I lkie the small adjustments in the introduction. We should keep those.

File: draft-ietf-taps-arch.md [hunk 5]
```diff
@@ -89,15 +97,21 @@ This document describes an architecture for exposing transport protocol features
...
+TCP has scaled remarkably well over the past three and a half decades, but its total ubiquity has hidden an uncomfortable fact: the network is not really a	file, and stream abstractions are too simplistic for many modern application	
+programming models.	
+
+In the meantime, the nature of Internet access, and the variety of Internet	transport protocols, is evolving. The challenges that new protocols and access paradigms present to the Socket API and to programming models based on them	
+inspire the design principles of a new approach.
+Therefore, the new Transport Services architecture is based on the survey of services provided by IETF transport protocols and congestion control mechanisms {{?RFC8095}}, and the distilled minimal set of the features offered by transport protocols {{?I-D.ietf-taps-minset}}. These documents identified common features and patterns across all transport protocols developed thus far in the IETF.
```
Comment (abrunstrom) [inline draft-ietf-taps-arch.md]: I am fine leaving this extension out and keeping it simple. If we do decide to keep the text. Then I would like "new Tarnsport Services architecture" to be just "Transport Services architecture. I do not like the "new" for reasons previously commented on.

**Pull request merged**